mcp-server/src/index.ts

```diff
@@ -0,0 +1,775 @@
+      inputSchema: zodToJsonSchema(batchInputSchema),
+    },
+    {
+      name: 'compare_repositories',
+      description: 'Compare multiple repositories across key metrics with intelligent analysis',
+      inputSchema: zodToJsonSchema(compareReposSchema),
+    },
+    {
+      name: 'analyze_trends',
+      description: 'Perform comprehensive trend analysis on metrics over time',
+      inputSchema: zodToJsonSchema(trendAnalysisSchema),
+    },
+    {
+      name: 'get_ecosystem_insights',
+      description: 'Get ecosystem-level insights for languages, topics, or organizations',
+      inputSchema: zodToJsonSchema(ecosystemInsightsSchema),
+    },
+    {
+      name: 'server_health',
+      description: 'Get server health status, cache statistics, and performance metrics',
+      inputSchema: zodToJsonSchema(healthCheckSchema),
+    }
+  ],
+}));
+
+
+server.setRequestHandler(CallToolRequestSchema, async (request: any) => {
+  try {
+    if (!request.params.arguments) {
+      throw new Error("Arguments are required");
+    }
+
+    switch (request.params.name) {
+      case 'get_open_digger_metric': {
+        const args = inputSchema.parse(request.params.arguments);
+        const url = buildUrl(args);
+        const data = await fetchWithCache(url, DEFAULT_TTL_SECONDS);
+        
+        return { 
+          content: [{ 
+            type: 'text', 
+            text: JSON.stringify({
+              data,
+              metadata: {
+                metric: args.metricName,
+                entity: `${args.owner || args.login}${args.repo ? '/' + args.repo : ''}`,
+                platform: args.platform,
+                timestamp: new Date().toISOString(),
+                cached: true // Will be determined by fetchWithCache
+              }
+            }, null, 2) 
+          }] 
+        };
+      }
+
+      case 'get_open_digger_metrics_batch': {
+        const args = batchInputSchema.parse(request.params.arguments);
+        const results = [];
+        
+        // Process in batches to respect API limits
+        for (let i = 0; i < args.requests.length; i += BATCH_SIZE) {
+          const chunk = args.requests.slice(i, i + BATCH_SIZE);
+          const chunkResults = await Promise.all(chunk.map(async (r: z.infer<typeof inputSchema>) => {
+            try {
+              const url = buildUrl(r);
+              const data = await fetchWithCache(url, DEFAULT_TTL_SECONDS);
+              return { ok: true, request: r, data };
+            } catch (err) {
+              return { ok: false, request: r, error: (err as Error).message };
+            }
+          }));
+          
+          results.push(...chunkResults);
+          
+          // NOTE: Add delay between batches to avoid rate limits
+          // TODO: We need to test & confirm if this is necessary  |  cc: @birdflyi, @frank-zsy
+          if (i + BATCH_SIZE < args.requests.length) {
```

Comment:
Unnecessary branching conditions.

Reply:
Thanks for flagging! I'll get this resolved.